src/steps/email-field-validation.ts

```diff
@@ -0,0 +1,104 @@
+import { Step, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+import { Email, Inbox } from '../models';
+
+export class EmailFieldValidationStep extends BaseStep implements StepInterface {
+  private operators: string[] = ['should contain', 'should not contain', 'should be'];
+
+  protected stepName: string = 'Check a field on a Mailgun Email';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'the (?<field>(subject|body-html|body-plain|from)) of the (?<position>\\d+)(?:(st|nd|rd|th))? mailgun email for (?<email>.+) (?<operator>(should contain|should not contain|should be)) (?<expectation>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'email',
+    type: FieldDefinition.Type.EMAIL,
+    description: 'The inbox\'s email address',
+  }, {
+    field: 'field',
+    type: FieldDefinition.Type.STRING,
+    description: 'Field name to check',
+  }, {
+    field: 'expectation',
+    type: FieldDefinition.Type.ANYSCALAR,
+    description: 'Expected field value',
+  }, {
+    field: 'position',
+    type: FieldDefinition.Type.NUMERIC,
+    description: 'The nth message to check from the email\'s inbox',
+  }, {
+    field: 'operator',
+    type: FieldDefinition.Type.STRING,
+    description: 'The operator to use when performing the validation. Current supported values are: should contain, should not contain, and should be',
+  }];
+
+  async executeStep(step: Step) {
+    const stepData: any = step.getData() ? step.getData().toJavaScript() : {};
+    const expectation = stepData.expectation;
+    const field = stepData.field;
+    const position = stepData.position || 1;
+    const operator = stepData.operator;
+
+    try {
+      const inbox: Inbox = await this.client.getInbox(stepData.email);
+
+      if (!inbox || inbox === null) {
+        return this.error('Cannot fetch inbox for: %s', [
+          stepData.email,
+        ]);
+      }
+
+      if (!inbox.items[position - 1]) {
+        return this.error('Cannot fetch email in position: %s', [
+          position,
+        ]);
+      }
+
+      const storageUrl: string = inbox.items[position - 1].storage.url;
+      const email: Email = await this.client.getEmailByStorageUrl(storageUrl);
+
+      if (email === null || !email) {
+        return this.error('Cannot fetch email in position: %s', [
+          position,
+        ]);
+      }
+
+      if (this.executeComparison(expectation, email[field], operator)) {
+        return this.pass('Expected value %s %s %s', [
+          expectation,
+          operator,
+          email[field],
+        ]);
+      } else {
+        return this.fail('Comparison failed using: %s operator. Actual: %s Expected: %s', [
+          operator,
+          email[field],
+          expectation,
+        ]);
+      }
+    } catch (e) {
+      return this.error('There was an error reaching mailgun: %s', [e.toString()]);
```

Comment:
- [ ] My guess is that the error would contain enough context here for the user to diagnose, no need to name drop mailgun.  Perhaps: `There was an error retrieving email messages: %s`